### DISCUSSION TITLE
I could use some help building Glances from source

### DISCUSSION THREAD

**Discussion by LasseJacobs** [nicolargo/glances#1892] [Q&A] [answered]
Hi, 

I am trying to build glances from source and play around with it a bit. I tried building the image using the Makefile but ran into some issues on my system. I solved a few (related to my setup, pip wasn't working, node version was to high,...) then I got to the point where it shows me this error:
```
  File "/Users/lassejacobs/Projects/Adhoc-Repos/glances/venv/lib/python3.8/site-packages/psutil/__init__.py", line 1859, in cpu_freq
    ret = _psplatform.cpu_freq()
  File "/Users/lassejacobs/Projects/Adhoc-Repos/glances/venv/lib/python3.8/site-packages/psutil/_psosx.py", line 180, in cpu_freq
    curr, min_, max_ = cext.cpu_freq()
FileNotFoundError: [Errno 2] No such file or directory (originated from sysctlbyname('hw.cpufrequency'))
make: *** [run] Error 1
```
I think this might be related to the fact that I am on an M1 Mac... So I SSH'ed into my linux server to try on x86 linux. I wanted to try building using the docker file you provide, that worked great. Using to build:
`docker build --target dev -f docker-files/debian.Dockerfile -t glances .`
And then:
`docker run -d --restart="always" -p 61208-61209:61208-61209 -e GLANCES_OPT="-w" -v /var/run/docker.sock:/var/run/docker.sock:ro --pid host glances:latest`
to run the image I built. Using this I was able to connect to the webserver from my laptop and everything seems to work. 

However, then I made some modifications to the html files in glances/output/static, rebuilt, and re-ran the image. Yet when I open the web page I don't see my changes reflected. I looked at the source for the docker file and see:
`RUN pip3 install --no-cache-dir --user glances[all]`

Does that mean it install everything from the PIP repository rather than using my local files? (I am quite new to the python ecosystem) I there any way I can build the image using my local files? Or how would you recommend building from source? 

I am mostly interested in the project for learning purposes, this project seemed really cool and with reasonable complexity. 

Thanks in advance!

**Comment by nicolargo** [comment, marked as answer]
Hi @LasseJacobs 

Maybe the issue is related to https://github.com/giampaolo/psutil/issues/1892

Not sure that PsUtil can be build on Apple M1 for the moment...

Nicolas

**Reply by LasseJacobs** [reply]
Hi @nicolargo,
Thank you for reaching out some quickly. I indeed think it is not possible on M1 for the moment but that is ok. I am building it on my server now which is x86. But I have a problem that I don't see the changes I make in the UI reflected when I run the Glances. 

For example, I changed the following file: glances/glances/outputs/static/js/components/plugin-network/view.html
I added an 'X' to the title. The line now reads:
`<div class="table-cell text-left title">NETWORK X</div>`

~~However when I build the docker file and run it (using the commands I mention in the original post), it does not display the change. I still just says 'NETWORK'.~~

EDIT: Solved! Safari cached the page... I'm still new to Apple, I thought I cleared my cache a few times already but apparently not. Everything works as expected now! Going to have some fun with the code now :)

Any tips? I would like to change something in the UI, run it and see the result.

Thanks!
